### PR TITLE
Strided array doc: Remove reference to getindex

### PR DESCRIPTION
**Pull request by mbauman**
As noticed by Jutho, we don't assume that `getindex(A, i)` is the same as accessing the `i`th element as computed by the strides.

### CODE REVIEW

**Comment by Jutho** [comment]
Looks good to me.

As mentioned elsewhere (I am not trying to hijack this PR, though I am slightly failing at that), it would be great if there was actually a Julian way (i.e. not `Base.unsafe_wrap(Array, pointer(A), size(A))[i]` ) to do grab an element at a memory position `i` computed from the strides.

**Review by Jutho — COMMENTED** [review]
consecuitive -> consecutive

File: doc/src/manual/arrays.md [hunk 1]
```diff
@@ -716,12 +716,12 @@ create a `SubArray` view instead.
...
-index of dimension `k` by `1` should increase the index `i` of [`getindex(A,i)`](@ref) by
-[`stride(A,k)`](@ref). If a pointer conversion method [`Base.unsafe_convert(Ptr{T}, A)`](@ref) is
-provided, the memory layout must correspond in the same way to these strides. `DenseArray` is a
+external C and Fortran functions that expect this memory layout. Strided arrays must define a
+pointer conversion method [`Base.unsafe_convert(Ptr{T}, A)`](@ref) and a [`strides(A)`](@ref)
+method to define the memory layout. A provided [`stride(A, k)`](@ref) method accesses the `k`th
+element within this tuple. Increasing the
+index of dimension `k` by `1` should increase `i` such that it's no longer accessing the `i`th
+consecutive element from the base pointer, but rather the `i+stride(A,k)`th element. `DenseArray` is a
 very specific example of a strided array where the elements are arranged contiguously, thus it
```
Comment (Jutho) [inline doc/src/manual/arrays.md]: Alternative (but I am not a computer scientist nor a native speaker):
" Increasing the index of dimension `k` by `1` corresponds to a jump in memory that is the equivalent of `stride(A,k)` consecutive elements. Note that not all of these intermediate elements might be valid entries that belong to array `A` (in particular, when `stride(A,k) != `size(A,k-1)*stride(A,k-1)`)."

**Review by Jutho — APPROVED** [review]

**Comment by ViralBShah** [comment]
@mbauman Do we want to get this in?

**Comment by mbauman** [comment]
Yes, it's still good to remove `getindex` from this description.  It'd be nice to get someone else's read on if this wording is widely understandable.

**Comment by Jutho** [comment]
Rereading the new version, I notice that reference is made to some variable `i` while it is not defined what this is (one could infer it from the context but I still find it somewhat confusing).

An attempt of my own: 
```
Strided arrays must define a pointer conversion method
[`Base.unsafe_convert(Ptr{T}, A)`](@ref) that returns a pointer
to the memory location of the first element, and a [`strides(A)`](@ref)
method that defines the memory layout by returning a tuple of strides.
A [`stride(A, k)`](@ref) method to access the `k`th element within
this tuple is then automatically provided. Increasing the index of dimension `k` 
by `1` implies that the memory offset of the corresponding element with 
respect to  the base pointer is increased by `stride(A,k)`. `DenseArray` is
a very specific example of a strided array where the elements are arranged
contiguously, with thus `stride(A, 1) = 1` and 
`stride(A, k) = stride(A, k-1)*size(A, k-1)`. It provides its subtypes with the 
appropriate definition of strides. 
```

**Comment by mbauman** [comment]
Well I ended up completely re-writing this because I simply found what I originally wrote to be rather unreadable.  With the StridedArray interface chapter I don't think this needs to be nearly as legalistic as it once was — and I didn't find the QR example very compelling as it's not obvious that it actually is using LAPACK.

**Review by timholy — APPROVED** [review]
Very nice change.

At the end, would it be worth adding something like this?

"It is worth emphasizing that strides are about offsets in memory rather than indexing. If you are looking to convert between linear (single-index) indexing and cartesian (multi-index) indexing, see [`LinearIndices`](@ref) and [`CartesianIndices`](@ref)."

**Pull request merged**